doc/userguide/output/eve/eve-json-format.rst

```diff
@@ -1287,3 +1287,315 @@ Example of RFB logging, with full VNC style authentication parameters:
+
+Example of MQTT SUBSCRIBE logging:
+
+::
+
+  "mqtt": {
+    "message_type": "SUBSCRIBE",
+    "qos": 1,
+    "retain": false,
+    "dup": false,
+    "subscribe": {
+      "message_id": 1,
+      "topics": [
+        {
+          "topic": "topicX",
+          "qos": 0
+        },
+        {
+          "topic": "topicY",
+          "qos": 0
+        }
+      ]
+    }
+  }
+
+MQTT SUBACK fields
+~~~~~~~~~~~~~~~~~~
+
+* "suback.message_id": Original message ID this message refers to.
+* "suback.qos_granted": Array of QOS levels granted for the subscribed topics, in the order of the original request.
+* "suback.properties": (Optional, MQTT 5.0) SUBACK properties set on this request. See `3.9.2.1 in the spec <https://docs.oasis-open.org/mqtt/mqtt/v5.0/os/mqtt-v5.0-os.html#_Toc3901174>`_ for more information on SUBACK properties.
+
+Example of MQTT SUBACK logging:
+
+::
+
+  "mqtt": {
+    "message_type": "SUBACK",
+    "qos": 0,
+    "retain": false,
+    "dup": false,
+    "suback": {
+      "message_id": 1,
+      "qos_granted": [
+        0,
+        0
+      ]
+    }
+  }
+
+MQTT UNSUBSCRIBE fields
+~~~~~~~~~~~~~~~~~~~~~~~
+
+* "unsubscribe.message_id": (Only present if QOS level > 0) Message ID for this unsubscribe action.
+* "unsubscribe.topics": Array of topics to be unsubscribed from.
+* "unsubscribe.properties": (Optional, MQTT 5.0) UNSUBSCRIBE properties set on this request. See `3.10.2.1 in the spec <https://docs.oasis-open.org/mqtt/mqtt/v5.0/os/mqtt-v5.0-os.html#_Toc3901182>`_ for more information on UNSUBSCRIBE properties.
+
+Example of MQTT UNSUBSCRIBE logging:
+
+::
+
+  "mqtt": {
+    "message_type": "UNSUBSCRIBE",
+    "qos": 1,
+    "retain": false,
+    "dup": false,
+    "unsubscribe": {
+      "message_id": 1,
+      "topics": [
+        "topicX",
+        "topicY"
+      ]
+    }
+  }
+
+MQTT UNSUBACK fields
+~~~~~~~~~~~~~~~~~~~~
+
+* "unsuback.message_id": Original message ID this message refers to.
+
+Example of MQTT UNSUBACK logging:
+
+::
+
+  "mqtt": {
+    "message_type": "UNSUBACK",
+    "qos": 0,
+    "retain": false,
+    "dup": false,
+    "unsuback": {
+      "message_id": 1
+    }
+  }
+
+MQTT AUTH fields (MQTT 5.0)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+* "auth.reason_code": Return code/reason code for this message. See `3.15.2.1 in the spec <https://docs.oasis-open.org/mqtt/mqtt/v5.0/os/mqtt-v5.0-os.html#_Toc3901220>`_ for more information on these codes.
+* "auth.properties": (Optional, MQTT 5.0) Properties set on this request. See `3.15.2.2 in the spec <https://docs.oasis-open.org/mqtt/mqtt/v5.0/os/mqtt-v5.0-os.html#_Toc3901221>`_ for more information on these properties.
+
+Example of MQTT AUTH logging:
+
+::
+
+  "mqtt": {
+    "message_type": "AUTH",
+    "qos": 0,
+    "retain": false,
+    "dup": false,
+    "auth": {
+      "reason_code": 16
```

Comment:
Yes, but they vary between message types. Do you think it would be a good idea to also have a human-readable version in the output? I wasn't so sure because that would inflate the output quite a bit, and for just for doing searches on log output the integer codes would work as well (see examples in documentation).

Reply:
Ok, maybe leave a comment about this

Reply:
In the RST documentation? I have already mentioned that one should refer to the specification for the meaning of the reason code values (https://github.com/OISF/suricata/pull/4934/files#diff-7823d9d82df45a9858e8925a6259022eR1438, https://github.com/OISF/suricata/pull/4934/files#diff-7823d9d82df45a9858e8925a6259022eR1563 etc.).

Reply:
Thanks, that seems fair enough :-)